packages/query-core/src/queryObserver.ts

```diff
@@ -680,6 +680,9 @@ export class QueryObserver<
       if (this.options.throwOnError) {
         includedProps.add('error')
       }
+      if (includedProps.has('promise')) {
+        includedProps.add('data')
+      }
```

Comment:
> if users use the `promise` returned from useQuery, they are actually interested in the `data` it unwraps to.

if you `use(query.promise)`, you don't necessarily care about `query.data` and shouldn't need to re-render the whole tree

I think this should be deleted

```suggestion
```

I even added a test for this exactly this:

https://github.com/TanStack/query/blob/d1184babd1ba570da09798cad0e8613b7c5b69c8/packages/react-query/src/__tests__/useQuery.test.tsx#L6653-L6654

Reply:
hmm, I was thinking about this situation:

```
const { promise } = useQuery(...)
const data = use(promise)

return <div>data</div>
```

now, if an automatic refetch occurs (e.g. window focus, or calling `queryClient.refetchQuery` in a sibling component) and that refetch yields new data, the component wouldn't be informed about that _unless_ it also gets a new promise. Is it guaranteed to get a new promise? If so we can remove that code again.

Reply:
okay nice, then just revert my commit 🙃

Reply:
i reverted and added a test case for the above